jupyterlab_git/git.py

```diff
@@ -311,73 +313,70 @@ def __init__(self, contents_manager):
 
     async def detailed_log(self, selected_hash, current_path):
         """
-        Execute git log -1 --stat --numstat --oneline command (used to get
+        Execute git log -1 --numstat --oneline -z command (used to get
         insertions & deletions per file) & return the result.
         """
-        cmd = ["git", "log", "-1", "--stat", "--numstat", "--oneline", selected_hash]
+        cmd = ["git", "log", "-1", "--numstat", "--oneline", "-z", selected_hash]
         code, my_output, my_error = await execute(
             cmd, cwd=os.path.join(self.root_dir, current_path),
         )
 
         if code != 0:
             return {"code": code, "command": " ".join(cmd), "message": my_error}
 
+        total_insertions = 0
+        total_deletions = 0
         result = []
-        note = [0] * 3
-        count = 0
-        temp = ""
-        line_array = my_output.splitlines()
-        length = len(line_array)
-        INSERTION_INDEX = 0
-        DELETION_INDEX = 1
-        MODIFIED_FILE_PATH_INDEX = 2
-        if length > 1:
-            temp = line_array[length - 1]
-            words = temp.split()
-            for i in range(0, len(words)):
-                if words[i].isdigit():
-                    note[count] = words[i]
-                    count += 1
-            for num in range(1, int(length / 2)):
-                line_info = line_array[num].split(maxsplit=2)
-                words = line_info[2].split("/")
-                length = len(words)
-                result.append(
-                    {
-                        "modified_file_path": line_info[MODIFIED_FILE_PATH_INDEX],
-                        "modified_file_name": words[length - 1],
-                        "insertion": line_info[INSERTION_INDEX],
-                        "deletion": line_info[DELETION_INDEX],
-                    }
-                )
-
-        if note[2] == 0 and length > 1:
-            if "-" in temp:
-                exchange = note[1]
-                note[1] = note[2]
-                note[2] = exchange
+        line_iterable = iter(strip_and_split(my_output)[1:])
+        for line in line_iterable:
+            insertions, deletions, file = line.split('\t')
+
+            if file == '':
+                # file was renamed or moved
+                # we need next two lines of output
+                from_path = next(line_iterable)
+                to_path = next(line_iterable)
+                modified_file_name = from_path + " ⮕ " + to_path
```

Comment:
I'm not sure introducing unicode symbol here is the best plan, I would just use `=>` like the git cli does.

Reply:
```suggestion
                modified_file_name = from_path + " => " + to_path
```

Although in general I support any innovation in arrow rendering (at least those that make the arrows prettier), unicode in source should sadly still be avoided for the moment, since many tools (eg github) still can't correctly render/use old fonts that don't contain many glyphs